vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -493,6 +505,517 @@ protected void fireUpdateProgress(long totalBytes, long contentLength) {
         fireEvent(new ProgressUpdateEvent(this, totalBytes, contentLength));
     }
 
+    @DomEvent("file-reject")
+    public static class FileRejectEvent extends ComponentEvent<Upload> {
```

Comment:
IMO we should try to remove all of these:
- Even when used internally, it looks like the event classes and the add listener method are not really necessary, and can be easily removed / replaced
- There is a set of "public" event classes in the same package, with very similar naming (`FileRejectEvent` vs `FileRejectedEvent`) which is confusing
- A lot of these events don't really work on the server-side (`UploadBeforeEvent`)

Of the generated event listeners / events only the following are used at the moment:

- `addFileRejectedListener` - Just maps the generated event to the public event, so we could just use the public event instead: https://github.com/vaadin/flow-components/blob/acfc30193bd450505917c88a41de79077b5ef575/vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java#L118-L119

- `addUploadAbortListener` - Does not use the event, could be replaced with `getElement().addEventListener`: https://github.com/vaadin/flow-components/blob/acfc30193bd450505917c88a41de79077b5ef575/vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java#L122

- `addUploadErrorListener` - Does not use the event, could be replaced with `getElement().addEventListener`: https://github.com/vaadin/flow-components/blob/acfc30193bd450505917c88a41de79077b5ef575/vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java#L111-L113

- `addUploadStartListener` - Does not use the event, could be replaced with `getElement().addEventListener`: https://github.com/vaadin/flow-components/blob/acfc30193bd450505917c88a41de79077b5ef575/vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java#L145

- `addUploadSuccessListener` - Does not use the event, could be replaced with `getElement().addEventListener`: https://github.com/vaadin/flow-components/blob/acfc30193bd450505917c88a41de79077b5ef575/vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java#L115-L116

Reply:
Thank you, this really makes a lot of sense to me. Updated the PR accordingly.